### PR TITLE
Fix modelbit component and support boolean WyvernFeature

### PR DESCRIPTION
**Pull request by wintonzheng**
1. The modelbit request schema we're enforcing will be the [single argument format](https://doc.modelbit.com/deployments/rest-api/batch-inference#request-format), where the single argument will be a list of feature values.
2. support boolean in our FeatureMap

- [x] Does this PR have impact on local development experience? If yes, make sure you have a plan and add the documentations to address issues that come with the change
- [ ] bump version
- [ ] make a release
- [ ] publish to pypi service

### CODE REVIEW

File: wyvern/components/models/modelbit_component.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def __init__(
         self._auth_token = auth_token or self.AUTH_TOKEN
         self._modelbit_url = url or self.URL
         self.headers = {
-            "Authorization": f"Bearer {self._auth_token}",
```
Comment (wintonzheng) [inline wyvern/components/models/modelbit_component.py:52]: according to modelbit documentation: https://doc.modelbit.com/deployments/rest-api/api-keys

`  -H "Authorization: YOUR_API_KEY"`

File: wyvern/components/models/modelbit_component.py [hunk 2]
```diff
@@ -87,12 +87,10 @@ async def build_requests(
         all_requests = [
             [
                 idx + 1,
-                {
-                    "features": [
-                        self.get_feature(identifier, feature_name)
-                        for feature_name in self.modelbit_features
-                    ],
-                },
+                [
```
Comment (wintonzheng) [inline wyvern/components/models/modelbit_component.py:90]: the default input is going to be the feature vector/list.

File: wyvern/wyvern_typing.py [hunk 3]
```diff
@@ -20,4 +20,4 @@
 REQUEST_SCHEMA = TypeVar("REQUEST_SCHEMA", bound=BaseModel)
 RESPONSE_SCHEMA = TypeVar("RESPONSE_SCHEMA", bound=BaseModel)
 
-WyvernFeature = Union[float, str, List[float], None]
+WyvernFeature = Union[bool, int, float, str, List[float], None]
```
Comment (wintonzheng) [inline wyvern/wyvern_typing.py:23]: There might be a breaking change:
currently the logged boolean features are either `1.0` or `0.0`. with this change, it'll become true/false

Comment (wintonzheng) [reply wyvern/wyvern_typing.py:23]: @suchintan need your input.

Comment (suchintan) [reply wyvern/wyvern_typing.py:23]: Should we update the logging code to re-cast those to 1.0 or 0.0?

Comment (wintonzheng) [reply wyvern/wyvern_typing.py]: one more fix: we should just have float but not int!!!
tested it out:

```
In [1]: from wyvern.entities.feature_entities import FeatureData

In [2]: from wyvern.entities.identifier import Identifier

In [3]: identifier = Identifier(identifier="dfdadf", identifier_type="product")

In [4]: fd = FeatureData(identifier=identifier, features={"a": 5, "dafdas": True, "blabla": 4
   ...: .2})

In [5]: fd
Out[5]: identifier=product::dfdadf features={'a': 5.0, 'dafdas': True, 'blabla': 4.2}

In [6]: fd.json()
Out[6]: '{"identifier": {"identifier": "dfdadf", "identifier_type": "product"}, "features": {"a": 5.0, "dafdas": true, "blabla": 4.2}}'
```

File: wyvern/entities/request.py [hunk 4]
```diff
@@ -9,8 +11,17 @@ class BaseWyvernRequest(WyvernDataModel):
     request_id: str
     include_events: Optional[bool] = False
 
+    _identifier: Identifier = PrivateAttr()
```
Comment (wintonzheng) [inline wyvern/entities/request.py:14]: we're adding these interfaces for BaseWyvernRequest so that it behaves exactly the same as WyvernEntity until we actually migrate BaseWyvernRequest to inherit WyvernEntity (which will be done in a separate PR.)

File: wyvern/wyvern_typing.py [hunk 5]
```diff
@@ -20,4 +20,4 @@
 REQUEST_SCHEMA = TypeVar("REQUEST_SCHEMA", bound=BaseModel)
 RESPONSE_SCHEMA = TypeVar("RESPONSE_SCHEMA", bound=BaseModel)
 
-WyvernFeature = Union[float, str, List[float], None]
+WyvernFeature = Union[bool, float, str, List[float], None]
```
Comment (wintonzheng) [inline wyvern/wyvern_typing.py:23]: add the support for boolean in our feature map

File: wyvern/components/models/modelbit_component.py [hunk 6]
```diff
@@ -49,13 +49,13 @@ def __init__(
...
         self.headers = {
-            "Authorization": f"Bearer {self._auth_token}",
+            "Authorization": self._auth_token,
             "Content-Type": "application/json",
         }
 
         # TODO shu: test out the model_input_type
         self.model_input_type = self.get_type_args_simple(0)
-        self.model_ouput_type = self.get_type_args_simple(1)
+        self.model_output_type = self.get_type_args_simple(1)
```
Comment (wintonzheng) [inline wyvern/components/models/modelbit_component.py:58]: typo fix
